packages/gatsby/src/services/create-pages.ts

```diff
@@ -24,10 +24,30 @@ export async function createPages({
   activity.start()
   const timestamp = Date.now()
   const currentPages = new Map<string, IGatsbyPage>(store.getState().pages)
+
+  // Wrap the GraphQL function so we can measure how long it takes to run.
+  const originalGraphQL = gatsbyNodeGraphQLFunction
+  // eslint-disable-next-line
+  function wrappedGraphQL() {
+    const start = Date.now()
+    // @ts-ignore not sure how to type the following
+    const returnValue = originalGraphQL.apply(this, arguments) // eslint-disable-line
+    const end = Date.now()
+    const totalMS = end - start
```

Comment:
I am not sure if this is correct. `originalGraphQL` likely returns a promise. So the end of the query should be counted after the promise is resolved. Here is the return value of this function:

https://github.com/gatsbyjs/gatsby/blob/81401a5b771eb1557fa4b8b8fb756f457be7d223/packages/gatsby/src/bootstrap/create-graphql-runner.ts#L56-L61

Reply:
oh right! Thanks, will fix